src/signTx.c

```diff
@@ -120,11 +122,59 @@ static inline void advanceStage()
 	TRACE("Advancing sign tx stage to: %d", ctx->stage);
 }
 
-void respondSuccessEmptyMsg()
+// called from main state machine when a pool registration certificate
+// sub-machine is finished, or when other type of certificate is processed
+static inline void advanceCertificatesStateIfAppropriate()
 {
-	TRACE();
-	io_send_buf(SUCCESS, NULL, 0);
-	ui_displayBusy(); // needs to happen after I/O
+	TRACE("%u", ctx->stage);
+
+	if (ctx->stage == SIGN_STAGE_CERTIFICATES) {
```

Comment:
here I would likewise prefer switch on ctx->stage instead of assert in else branch